### PR TITLE
HW-355: Fixed activity and fixed header issues

### PR DESCRIPTION
**Pull request by mukeshcompucorp**
**Before:**
before-1

**After:**
after-1

### CODE REVIEW

File: ang/civicase/ActivityItemInner-communication.html [hunk 1]
```diff
@@ -1,51 +1,53 @@
-<a href class="act-task-checkbox" ng-if="activity.category.indexOf('task') > -1" ng-click="markCompleted(activity)" title="{{ activity.is_completed ? ts('Mark incomplete') : ts('Mark complete') }}">
-  <i class="fa" ng-class="{'fa-square-o': !activity.is_completed, 'fa-check-square-o': activity.is_completed}"></i>
-</a>
-<h4>
-  <span class="act-subject" ng-if="activity.subject">{{ activity.subject }}</span>
-  <span class="act-subject" ng-if="!activity.subject">{{ activity.type }}</span>
-  <span class="act-subject-draft" ng-if="activity.status_name === 'Draft'">{{ ts('Draft') }}</span>
-  <a class="act-star crm-i" ng-click="star(activity)" href ng-if="star && activity.is_star">
-    <i class="crm-i" ng-class="{'fa-star-o': activity.is_star === '0', 'fa-star': activity.is_star === '1'}" title="{{ activity.is_star === '0' ? ts('Add star') : ts('Unstar') }}"></i>
+<div class="communication-wrapper">
```
Comment (colemanw) [inline ang/civicase/ActivityItemInner-communication.html]: This should not be necessary. Every activity is already wrapped in a div with css classes corresponding to its category.

File: ang/civicase/ActivityItemInner-communication.html [hunk 2]
```diff
@@ -1,51 +1,53 @@
-<a href class="act-task-checkbox" ng-if="activity.category.indexOf('task') > -1" ng-click="markCompleted(activity)" title="{{ activity.is_completed ? ts('Mark incomplete') : ts('Mark complete') }}">
-  <i class="fa" ng-class="{'fa-square-o': !activity.is_completed, 'fa-check-square-o': activity.is_completed}"></i>
-</a>
```
Comment (colemanw) [inline ang/civicase/ActivityItemInner-communication.html]: Please don't remove this. A communication activity could also be a task.

File: ang/civicase/ActivityItemInner-communication.html [hunk 3]
```diff
@@ -1,51 +1,53 @@
...
+    <i class="fa" ng-class="{'fa-square-o': !activity.is_completed, 'fa-check-square-o': activity.is_completed}"></i>
   </a>
-  <div ng-if="activity.file_id" class="act-attachment-menu btn-group btn-group-sm">
-    <button
-      type="button"
-      class="btn btn-default dropdown-toggle"
-      data-toggle="dropdown"
-      title="{{ activity.file_id.length === 1 ? ts('1 Attachment') : ts('%1 Attachments', {1: activity.file_id.length}) }}"
-      ng-click="getAttachments(activity)"
+  <h4>
```
Comment (colemanw) [inline ang/civicase/ActivityItemInner-communication.html]: According to the mockup, the activity target contact names belong here.

File: ang/civicase/ActivityItemInner-communication.html (hunk 1 above)
Comment (colemanw) [reply ang/civicase/ActivityItemInner-communication.html]: This whole template will already be wrapped in `<div class="act-category-communication, etc">`

File: ang/civicase/ActivityItemInner-communication.html (hunk 2 above)
Comment (mukeshcompucorp) [reply ang/civicase/ActivityItemInner-communication.html]: @colemanw  I am not removing it.
I have just added wrapper,` <div class="communication-wrapper">`
and moved `<span class="act-subject-draft" ng-if="activity.status_name === 'Draft'">{{ ts('(Draft)') }}</span>` on different position based on mockup. Have a look at final blob https://github.com/mukeshcompucorp/org.civicrm.civicase/blob/4fa5f15cc52d181405afd268b023eac2bc2f8f39/ang/civicase/ActivityItemInner-communication.html

Reason for adding wrapper. We have slightly different design for communication which needs another class to handle design of it. The main wrapper does the standard set of design. I need a gray background with an arrow for communication, see https://github.com/mukeshcompucorp/org.civicrm.civicase/blob/4fa5f15cc52d181405afd268b023eac2bc2f8f39/css/general.css#L181

**Pull request merged**